### ISSUE TITLE
Request: add scrollend event

### ISSUE THREAD

**Issue by stevelacy** [zippyui/react-datagrid#271]
Request: add event when the scrollbox has been scrolled to the bottom. 
This allows a request to the server for more data on a non-paginated datagrid.

```diff

@@ -204,6 +204,11 @@ module.exports = React.createClass({
  
          this.scrollTop = scrollTop
  
 +        if (this.props.onScrollEnd) {
 +          var endIndex = this.getRenderEndIndex(props, state)
 +          this.props.onScrollEnd({ scrollTop, endIndex })
 +        }
 +
          if (props.virtualRendering){
  
              var prevIndex        = this.state.startIndex || 0
```

**Comment by davertron** [comment]
I would say add an 'onScroll' event period (maybe a special param could be added for when the bottom is hit?). I recently was trying to implement something that was very difficult because the grid doesn't have any kind of 'onScroll' handler. It would be AWESOME if that onScroll handler got data about the current data in the view as well (i.e. an array of the currently visible data elements).

**Comment by stevelacy** [comment]
Yeah, I hacked a onScroll event to fire on a fork: https://github.com/stevelacy/react-datagrid/blob/master/lib/index.js#L208
It sends back the endindex, with which you can infer the current data (minus a limit).
Though it does not actually determine when you hit the bottom - as the measurement needs to be reset by scrolling up.

**Comment by davertron** [comment]
Hmm, yeah I was thinking after I posted that that if you just had a simple onScroll with the indices you could figure out if you had hit the bottom or not based on what data was visible, but that might be more complicated than necessary if the grid itself already has that info somewhere.
